crates/store/re_chunk_store/src/dataframe.rs

```diff
@@ -37,23 +37,29 @@ pub enum ColumnDescriptor {
     Control(ControlColumnDescriptor),
     Time(TimeColumnDescriptor),
     Component(ComponentColumnDescriptor),
+    DictionaryEncoded(ComponentColumnDescriptor),
```

Comment:
This reads like there are four types of columns, and `DictionaryEncoded` is neither a `Control`, `Time` or a `Component` column. But really, I believe it is just another `Component` column, just encoded differently.

We could introduce another layer here, like:

```rs
pub enum ColumnDescriptor {
    Control(ControlColumnDescriptor),
    Time(TimeColumnDescriptor),
    Component(ComponentColumnDescriptorThingy),
}

struct ComponentColumnDescriptorThingy {
    col_descr: ComponentColumnDescriptor,
    encoding: ColumnEncoding,
}

enum ColumnEncoding {
    ListArray, // or whatever non-dictionary-encoded component columns use
    Dictionary,
}
```

I think structurally that would be the cleanest, and would open up for more encoding options in the future.

---- 

I can understand if you think that's a bit overkill right now though, so if you prefer the current structure of the PR, at least name them better and add some docstring to explain them:

``` rs
pub enum ColumnDescriptor {
    /// A control column (`RowId`) encoded with…
    Control(ControlColumnDescriptor),

    /// A timeline column encoded with…
    Time(TimeColumnDescriptor),

    /// A component column encoded with…
    ListArrayComponent(ComponentColumnDescriptor),

    /// A component column encoded with…
    DictionaryComponent(ComponentColumnDescriptor),
```